android/src/main/java/com/onesignal/flutter/OneSignalUser.java

```diff
@@ -163,6 +165,13 @@ private void getTags(MethodCall call, Result result) {
         replySuccess(result, OneSignal.getUser().getTags());
     }
 
+    private void trackEvent(MethodCall call, Result result) {
+        String name = call.argument("name");
+        Map<String, Object> properties = call.argument("properties");
```

Comment:
Wondering if you tested booleans or ints as top-level values? Dot they translate properly? Such as:
```dart
var props = {'foo': 5, 'bar': true};
OneSignal.User.trackEvent(eventName, properties: props);
```